bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GenericItem.java

```diff
@@ -401,15 +403,44 @@ public StateDescription getStateDescription() {
 
     @Override
     public StateDescription getStateDescription(Locale locale) {
+        StateDescription fullStateDescription = null;
         if (stateDescriptionProviders != null) {
+            BigDecimal minimum = null;
+            BigDecimal maximum = null;
+            BigDecimal step = null;
+            String pattern = null;
+            boolean readOnly = false;
+            List<StateOption> options = null;
+            boolean found = false;
             for (StateDescriptionProvider stateDescriptionProvider : stateDescriptionProviders) {
                 StateDescription stateDescription = stateDescriptionProvider.getStateDescription(this.name, locale);
                 if (stateDescription != null) {
-                    return stateDescription;
+                    found = true;
+                    if (minimum == null) {
+                        minimum = stateDescription.getMinimum();
+                    }
+                    if (maximum == null) {
+                        maximum = stateDescription.getMaximum();
+                    }
+                    if (step == null) {
+                        step = stateDescription.getStep();
+                    }
+                    if (pattern == null) {
+                        pattern = stateDescription.getPattern();
+                    }
+                    if (stateDescription.isReadOnly()) {
+                        readOnly = true;
+                    }
+                    if (options == null) {
+                        options = stateDescription.getOptions();
+                    }
                 }
             }
+            if (found) {
+                fullStateDescription = new StateDescription(minimum, maximum, step, pattern, readOnly, options);
```

Comment:
Yes, that is exactly the idea. It gives us a chance to get options from channel for example. I could add a comment.
What are the unexpected results ?
Do you prefer I keep this method unchanged and i create another one with that code ?
My PR is clearly to fix two different things that are more or less linked to state description. Splitting in two PRs means more work, more time. But ok, I will take the time !